pkg/logs/sds/scanner.go

```diff
@@ -23,6 +24,21 @@ const ScannedTag = "sds_agent:true"
 
 const SDSEnabled = true
 
+var (
+	tlmSDSConfiguredRules = telemetry.NewCounterWithOpts("sds", "rules_configured", []string{},
+		"Number of configured rules.", telemetry.Options{DefaultMetric: true})
+	tlmSDSMalformedRules = telemetry.NewCounterWithOpts("sds", "rules_malformed", []string{},
+		"Number of malformed rules received through RC.", telemetry.Options{DefaultMetric: true})
```

Comment:
Is there any guidance on telemetry naming in the Agent to compare this with?

If we following the [Prometheus naming guidelines](https://prometheus.io/docs/practices/naming/) for example this should be `malformed_rules_total` (same with the others here). I'm not saying we should follow the Prometheus guidelines; I'm just wondering if there is another set that the Agent follows.

I'm similarly wondering if there are guidelines about when to use different metric names or the same metric name with tags. For example I could see us model rule application like:

```
sds_rules{state="disabled"}
sds_rules{state="malformed"}
```

etc.

Reply:
Not much that the Agent follows, however, the one quoted here would appear as `datadog.agent.sds.rules_malformed` in Datadog. This comes from using the `pkg/telemetry` of the Agent.

As for using tag instead, I thought about it but preferred to go that path for no reasons. No strong opinion on switching to using tag instead (which could help with `group by` clauses for visualisation).

Reply:
👍 Indeed, seems to be no standard for metric names within the Agent.

Agreed, I think using tags for the different rule states could make it easier to graph them altogether.